Migrate group show to Bootstrap

eduardoj opened the pull request:
Updating the group members is not done yet. It will be done in another pull request.

![2018-11-29-group_show](https://user-images.githubusercontent.com/24919/49242403-f1e58000-f40a-11e8-815b-7f2f8ea24d57.gif)

bgeuken commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+          %tr
+            %th Login Name
+        %tbody
+          - if @group.users.any?
```
To see that there is no content? :thinking:

I think I would find it confusing if the table some times is there and sometimes it's not.  Since that's the main job of this tab, showing the group members.

eduardoj replied:
This is how it will look like with the current code for a group without users:

![group_members_empty](https://user-images.githubusercontent.com/24919/49281286-79bf9e80-f48c-11e8-921d-1ede7fee4149.png)

I'm with @bgeuken.

dmarcoux replied:
I agree with @bgeuken and @eduardoj

Ana06 replied:
We do it differently in other pages such as the user home page page. We need some application consistency here.

dmarcoux commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+
+    .tab-pane#group-members{ aria: { controls: 'group-members' }, role: 'tabpanel' }
+      %h3 Group Members
+      %table.responsive.table.table-striped.table-bordered#group-members-table{ width: '100%' }
```
`width: 100%` can be replaced by the Bootstrap class `w-100`

eduardoj replied:
> `width: 100%` can be replaced by the Bootstrap class `w-100`

Done.

eduardoj commented on src/api/app/views/webui2/shared/_requests_table.html.haml:
```diff
@@ -10,4 +10,4 @@
       %th Priority
       %th
-%tbody
+  %tbody
```
> this is still wrong indented confused

Hm... I don't think so. Can you click on "View file" and check again?

Ana06 replied:
Github bug! :bug:

eduardoj commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+= content_for :ready_function do
+  :plain
+    $('#group-members-table').DataTable({
+      responsive: true
```
> why is the responsive necessary? :confused:

I used what stays in the [table page](https://obs-patterns.netlify.com/atoms/table.html) of our patterns, for a Table with DataTable.

Ana06 replied:
And I asked why it is needed (not where you copied from) :rofl: either you do it with the responsible class or with the parameter, but not both. It is an unnecessary redundancy (this should also be updated in the patterns, I'll take care) :unamused:

Ana06 replied:
https://datatables.net/reference/option/responsive#Default

Ana06 replied:
https://datatables.net/extensions/responsive/init

eduardoj replied:
Parameter removed. :+1:

dmarcoux commented:
To add to what @bgeuken, we also have the tabs in the [patterns](https://obs-patterns.netlify.com/atoms/tabs.html). Once you add the dropdown, it should be working as you already have the `collapsible` class on the `ul` tag.

eduardoj commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+- @pagetitle = "Group #{@group.title}"
```
> `.title` is not needed

Removed.

eduardoj commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+          - if @group.users.any?
+            - @group.users.each do |user|
+              %tr
+                %td= link_to(user.login, user_show_path(user))
```
> `.login` should be needed

If you wanted to say `should not be needed`, yes. Removed.

eduardoj commented:
@Ana06:

> from the first commit:
> 
> > from configuration to Bootstrap
> 
> what does this mean? :confused:

I removed it `from configuration` in the description of the pull request. I realized that could be confusing there, but forgot to remove it in the commit message. Removed from the commit message too.

> Was the second commit breaking something? I would say so. Can you include this information in the commit message?

It was breaking the indentation. I will add some more comment to clarify the what is th outputs in both cases.

> Tests need to be addapted

Sure. I'm working on it.

eduardoj commented:
@bgeuken:

> The tabs disappear on smaller screens. We had the same issue. You gone need a dropdown

Done. It was esasy! :+1:

eduardoj commented:
> > Tests need to be addapted
> 
> Sure. I'm working on it.

Done. I think I've addressed all the changes requested. @Ana06, @bgeuken, @dmarcoux please have another look.

Ana06 commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+          Edit group members
+
+= content_for :ready_function do
+  $('#group-members-table').DataTable();
```
I think may PR (https://github.com/openSUSE/open-build-service/pull/6410) will get merged before than yours, so you may want to use the new pattern: https://github.com/openSUSE/obs-patterns/pull/42

eduardoj replied:
I suppose that otherwise you will not approve the PR. Let's wait then.

eduardoj replied:
#6410 was merged. New pattern used.

Ana06 commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+
+    .tab-pane#group-members{ aria: { controls: 'group-members' }, role: 'tabpanel' }
+      %h3 Group Members
+      %table.responsive.table.table-striped.table-bordered.w-100#group-members-table
```
Drop repsosive, it is not working properly

eduardoj replied:
Done.

Ana06 commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,56 @@
+          Edit group members
+
+= content_for :ready_function do
+  initializeDataTable('#group-members-table');
```
you need `obs-dataTable` otherwise the fix doesn't work. Please check: https://github.com/openSUSE/obs-patterns/pull/42

eduardoj replied:
Done.

Ana06 replied:
The class is not in the correct place. Have you tried it? :confused:

dmarcoux commented on src/api/app/views/webui2/webui/groups/show.html.haml:
```diff
@@ -0,0 +1,57 @@
+              %tr
+                %td This group does not contain users.
+        - if User.current.is_admin? || @group.group_maintainers.where(user: User.current).exists?
+          = link_to group_edit_title_path(title: @group) do
```
This should also be handled to complete the card, but it can be in another PR.

dmarcoux replied:
I created a card so we don't forget it: https://trello.com/c/MW1IGcM7/987-edit-group-members
